launcher/ui/pages/instance/ServersPage.cpp

```diff
@@ -112,11 +125,42 @@ struct Server {
     bool m_checked = false;
     bool m_up = false;
     QString m_motd;  // https://mctools.org/motd-creator
-    int m_ping = 0;
-    int m_currentPlayers = 0;
+    std::optional<int> m_currentPlayers; // nullopt if not calculated/calculating
     int m_maxPlayers = 0;
 };
 
+class ServerPingTask : public Task {
+    Q_OBJECT
+  public:
+    explicit ServerPingTask(QObject* parent, Server &server) : Task(parent), m_server(server) {}
+    ~ServerPingTask() override = default;
+
+
+  protected:
+    virtual void executeTask() override {
+        qDebug() << "Querying status of " << m_server.m_address;
+        auto [domain, port] = m_server.splitAddress();
+        McResolver *resolver = new McResolver(nullptr, domain, port);
+        QObject::connect(resolver, &McResolver::succeed, [=](QString ip, int port) {
```

Comment:
please do not use `[=]` (capture all) in lambda

Reply:
Sure !

Could you explain to me why this shouldn't be used ? I thought `[=]` only mean "capture implicitly what is used", not "capture all"

Reply:
it leads to some situation where your using something implicitly you didn't intend to. like the `this` pointer, or a shared pointer to a qobject the signals bound to causing circular references.
it's just best practice with async lambdas to be explicit as possible about what's getting captured.